docs/architecture/adr/0012-angular-filename-convention.md

```diff
@@ -45,16 +45,27 @@ At Bitwarden we also use a couple of more types:
 - `.request` - Api Request
 - `.response` - Api Response
 - `.type` - Enum
-- `.service.abstraction` - Abstract class for a service, used for DI, not all services needs an
-  abstract class
 
 The class names are expected to use the suffix as part of their class name as well. I.e. a service
 implementation will be named `FolderService`, a request model will be named `FolderRequest`.
 
-In the event a service can't be fully implemented, an abstract class is created with the
-`Abstraction` suffix. This typically happens if the Angular and Node implementations have to differ
-for one reason or another. Traditionally interfaces would be used, but a TypeScript interface cannot
-be used to wire up dependency injection in JavaScript.
+## Abstract & Default Implementations
+
+The Bitwarden clients codebase serves multiple clients, one of which is node based and doesn't use
+the Angular Dependency Injection. In order to make code useable in both Angular and non Angular
+based clients we generally use abstract classes to define the interface. Ideally we would use
+_interfaces_ but TypeScript interfaces cannot be used to wire up dependency injection in JavaScript.
+
+All consumers will use the abstract class as a parameter in their constructor which will be manually
+wired up in the CLI and use Angular dependency injection in the Angular clients. In the case a
+dependency is only used in the Angular client, the abstract class can be omitted and the
+implementation referenced directly using `@Injectable`.
+
+To avoid naming conflicts, it's forbidden to name classes with the same name as the abstract class.
```

Comment:
💭 It isn't clear to me why a naming conflict is a problem. Why not instead use `import as` to avoid the conflict?

Note: this is not saying I disagree, only that this ADR does not assert why we should opt to use one over the other. Indeed, it seems out of place to talk about class naming conventions in an ADR concerning file naming conventions.

Reply:
There's nothing inherently wrong with name conflicts, it's more an area of confusion and additional overhead to consider when resolving automatic imports in editors, and performing code reviews.

The original reason this was part of the ADR was due to the initial iteration using a `.implementation` suffix, which later became `.abstraction`. The implicit effect of this rule is that the file will be named `default-`.

It's true this isn't directly related to this ADR and we could remove it and just keep it in the presentation document.

Reply:
I think it's important to include that justification in the ADR. As is, the PR reads as "Do it because I said so" when it's really "Do it because `import as` is considered harmful."